tests/test_single_rules.py

```diff
@@ -81,3 +81,49 @@ def test_forbid_combination(self):
                 f"DEVICE_COMPILER: {comb[1][0]} {comb[1][1]}, "
                 f"UBUNTU: {comb[2][1]}",
             )
+
+class TestCUDA11CLANGCUDA16UBUNTU2004(unittest.TestCase):
+    def setUp(self):
+        global param_map
+        param_map[HOST_COMPILER] = 0
+        param_map[DEVICE_COMPILER] = 1
+        param_map[BACKENDS] = 2
+
+    def setDown(self):
+        global param_map
+        param_map = {}
+
+    # test if CUDA 11.3 + clang-cuda 16 is disabled
+    def test_forbid_combination(self):
+        for cuda_version in ["11.0", "11.1", "11.2", "11.3", "11.4", "11.5"]:
+            comb = [(CLANG_CUDA, "16"), (CLANG_CUDA, "16"), (ALPAKA_ACC_GPU_CUDA_ENABLE, cuda_version)]
```

Comment:
The type is not correct. Backends needs to be a list, also if there is only one element.

` [(CLANG_CUDA, "16"), (CLANG_CUDA, "16"), [(ALPAKA_ACC_GPU_CUDA_ENABLE, cuda_version)]]`